tests/journald_tests.cpp

```diff
@@ -877,6 +879,184 @@ TEST_P(JournaldLoggerTest, ROOT_LogToJournald)
 }
 
 
+class FluentbitLoggerTest : public JournaldLoggerTest
+{
+public:
+  virtual void SetUp() override
+  {
+    // NOTE: We explicitly do not call the JournaldLoggerTest::SetUp function
+    // as we need to modify some module parameters before loading them.
+    MesosTest::SetUp();
+
+    // Create the mock fluentbit server.
+    Try<network::inet::Socket> server = network::inet::Socket::create();
+    ASSERT_SOME(server);
+
+    fluentbit_server = server.get();
+
+    Try<network::inet::Address> bind =
+      fluentbit_server->bind(
+          network::inet::Address(net::IP(process::address().ip), 0));
+    ASSERT_SOME(bind);
+
+    Try<Nothing> listen = fluentbit_server->listen(2);
+    ASSERT_SOME(listen);
+
+    // Read in the example `modules.json`.
+    Try<std::string> read =
+      os::read(path::join(MODULES_BUILD_DIR, "journald", "modules.json"));
+    ASSERT_SOME(read);
+
+    Try<JSON::Object> json = JSON::parse<JSON::Object>(read.get());
+    ASSERT_SOME(json);
+
+    Try<Modules> _modules = protobuf::parse<Modules>(json.get());
+    ASSERT_SOME(_modules);
+
+    modules = _modules.get();
+
+    // Replace the dummy values for `fluentbit_ip` and `fluentbit_port`
+    // with the server bind address above.
+    foreach (Modules::Library& library, *modules.mutable_libraries()) {
+      foreach (Modules::Library::Module& module, *library.mutable_modules()) {
+        if (module.has_name() && module.name() == JOURNALD_LOGGER_NAME) {
+          foreach (Parameter& parameter, *module.mutable_parameters()) {
+            if (parameter.key() == "fluentbit_ip") {
+              parameter.set_value(stringify(bind.get().ip));
+            } else if (parameter.key() == "fluentbit_port") {
+              parameter.set_value(stringify(bind.get().port));
+            }
+          }
+        }
+      }
+    }
+
+    // Initialize the modules.
+    Try<Nothing> result = ModuleManager::load(modules);
+    ASSERT_SOME(result);
+  }
+
+protected:
+  Option<network::inet::Socket> fluentbit_server;
+};
+
+
+INSTANTIATE_TEST_CASE_P(
+    LoggingMode,
+    FluentbitLoggerTest,
+    ::testing::Values(
+        std::string("fluentbit"),
+        std::string("fluentbit+logrotate")));
+
+
+// Sets up a TCP to mock the fluentbit component and then loads the
+// ContainerLogger module and runs a task.
```

Comment:
Description might additionally mention that the test expects a known log value to reach the Fluent Bit server.